lualib/lua_util.lua

```diff
@@ -1322,7 +1322,8 @@ exports.maybe_encrypt_header = function(header, settings, prefix)
     else
       encrypted_header = cryptobox:encrypt(header, nonce)
     end
-    return encrypted_header, nonce
+    local encrypted_header_with_nonce = nonce .. encrypted_header
+    return encrypted_header, nonce, encrypted_header_with_nonce
```

Comment:
Why do we need that? API should just return a string + nonce. It is a task for a caller to do the remaining.